sympy/printing/smtlib.py

```diff
@@ -205,11 +206,28 @@ def _print_BooleanFalse(self, x: BooleanFalse):
         return 'false'
 
     def _print_Float(self, x: Float):
-        f = x.evalf(self._precision) if self._precision else x.evalf()
-        return str(f).rstrip('0')
+        dps = prec_to_dps(x._prec)
+        str_real = mlib_to_str(x._mpf_, dps, strip_zeros=True, min_fixed=None, max_fixed=None)
+
+        if 'e' in str_real:
+            (mant, exp) = str_real.split('e')
+
+            if exp[0] == '+':
+                exp = exp[1:]
+
+            mul = self._known_functions[Mul]
+            pow = self._known_functions[Pow]
+
+            return r"(%s %s (%s 10 %s))" % (mul, mant, pow, exp)
+        elif str_real == "+inf":
+            return self._print_Function(float("inf")) # should throw error
+        elif str_real == "-inf":
+            return self._print_Function(-float("inf")) # should throw error
```

Comment:
Why are we not raising errors?

Reply:
I'm not sure if you are intending to mark this as todo list or going to keep it as intended.

Reply:
Good catch. This now throws an exception saying that infinite values are not supported in SMT. As far as I know, that's generally the case.